### ISSUE TITLE
Version flag should print “just” the version number

### ISSUE THREAD

**Issue by atifaziz** [filipw/dotnet-script#276]
Seems like #271 has implemented #263 but on the `-v`/`--version` flag.

It is conventional to have a version flag print _just_ the version number and nothing else so that it's machine-parsable for scripts that need to ensure version compatibility. It would be acceptable to have `--version` be the more verbose version that prints what is suggested for #263.

Note that `dotnet --version` does this too.

**Comment by seesharper** [comment]
I totally agree on this one and I actually started out with the `-i|--info` option until I realized that it conflicted with the existing `-i` option that means `interactive mode`
But, yes, `-v|--version` should ideally just print out the version number.

@filipw  Thoughts?

**Comment by atifaziz** [comment]
> So what you are saying is -v prints just the version number while --version prints the verbose output?

Yes, what I said is that it would be acceptable if you're not going for `--info` or `info` as a sub-command. Mind you, [Microsoft.Extensions.CommandLineUtils](https://www.nuget.org/packages/McMaster.Extensions.CommandLineUtils/) has a bug right now that's going to prevent that without a workaround. See natemcmaster/CommandLineUtils#94 for more details and you might want to chime in with your 2 cents. :wink:

**Comment by filipw** [comment]
> It is conventional to have a version flag print just the version

I think that's debatable. While many tools do that, many others print full version / runtime info (i.e. from the top of my head Mono, Java, g++, R, clang++). 
I'd like to keep the flags under reasonable control for now so would like to avoid having multiple flags for similar stuff or different output for different flag syntax.

Is this really a problem?

**Comment by seesharper** [comment]
Alternatively we can do `-e|--env` for displaying the environmental information that is now currently a part of `-v|--version`

**Comment by atifaziz** [comment]
> Is this really a problem?

It's never a problem until it is. :wink: I have written countless scripts in the past that had to check versions of installed programs and those that gave you a simple version string were a lifesaver. I'll admit this is way more of an issue on Windows than other platforms where there is a very limited tool-set to _grep_, _awk_ or _sed_ you way through the output from programs but putting that aside…

> While many tools do that, many others print full version / runtime info (i.e. from the top of my head Mono, Java, g++, R, clang++).

So my point was why not be in both camps? Do we have to pick a side? Just help the user!

I think the idea of machine/human-readable output (and sometimes both) from CLI programs (plain-text can be a protocol too) has not caught on entirely in the .NET community.

Here's a simple example of `git status`:

```
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   src/Dotnet.Script.Tests/Dotnet.Script.Tests.csproj
	modified:   src/Dotnet.Script/Dotnet.Script.csproj
	modified:   src/Dotnet.Script/Program.cs

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	src/Dotnet.Script/Properties/launchSettings.json

no changes added to commit (use "git add" and/or "git commit -a")
```

That's great for humans but you can also get a machine-readable version via `git status --procelain`:

```
 M src/Dotnet.Script.Tests/Dotnet.Script.Tests.csproj
 M src/Dotnet.Script/Dotnet.Script.csproj
 M src/Dotnet.Script/Program.cs
?? src/Dotnet.Script/Properties/launchSettings.json
```

Here's what the [help for that option](https://git-scm.com/docs/git-status/2.17.0#git-status---porcelainltversiongt) has to say:

> `--porcelain[=<version>]`
> Give the output in an easy-to-parse format for scripts. This is similar to the short output, but will remain stable across Git versions and regardless of user configuration. See below for details.
>
> The version parameter is used to specify the format version. This is optional and defaults to the original version v1 format.

There are a number of ways to do this for `--version`. Either have a long and short form, like `-v` and `--version`, or you can combine with other flags as follows:

- Option 1:
  - `--version`: prints just the version number (for scripts)
  - `--version` and `--verbose`: prints the longer version information to whatever detail you like (for humans)
- Option 2:
  - `--version`: prints the longer version information to whatever detail you like (for humans)
  - `--version` and `-q`/`--quiet`: prints just the version number (for scripts)

Option 1 allows you to reserve `-v` as an alias for `--verbose`.

**Comment by seesharper** [comment]
Fixed by #282

**Comment by atifaziz** [comment]
@seesharper #282 comes close but as I pointed out in [a comment of my review](https://github.com/filipw/dotnet-script/pull/282#discussion_r186107593), `dotnet script --version` prints a blank line before the version number:

```

0.24.0
```

It's close because we now have just the version number but the blank line requires unnecessary tricks like piping to `tail -n 1` to get the right result.
